src/ring_theory/roots_of_unity.lean

```diff
@@ -863,6 +863,78 @@ begin
       not_false_iff, one_ne_zero] }
 end
 
+/-- If `m : ℕ` is coprime with `n`,
+then the minimal polynomials of a primitive `n`-th root of unity `μ`
+and `μ ^ m` of are the same. -/
+lemma minimal_polynomial_eq_pow_coprime {m : ℕ} (hcop : nat.coprime m n) :
+  minimal_polynomial (is_integral h hpos) = minimal_polynomial
+  (is_integral (h.pow_of_coprime m hcop) hpos) :=
+begin
+  revert n hcop,
+  refine (unique_factorization_monoid.induction_on_prime m _ _ _),
+  { intros n hn h hpos,
+    simp only [nat.coprime_zero_left] at hn,
+    subst hn,
+    simp only [is_primitive_root.one_right_iff] at h,
+    subst h,
+    simp only [one_pow] },
+  { intros u hunit n hcop h hpos,
+    simp only [nat.is_unit_iff] at hunit,
+    subst hunit,
+    simp only [pow_one] },
+  { intros a p ha hprime hind n hcop h hpos,
+    replace hind := hind (nat.coprime.coprime_mul_left hcop) h hpos,
+    rw [hind],
+    replace hprime := nat.prime_iff_prime.2 hprime,
+    have hdiv := (nat.prime.coprime_iff_not_dvd hprime).1 (nat.coprime.coprime_mul_right hcop),
+    letI : fact p.prime := hprime,
+    rw [minimal_polynomial_eq_pow
+      (h.pow_of_coprime a (nat.coprime.coprime_mul_left hcop)) hpos hdiv],
+    have expeq : (μ ^ a) ^ p = μ ^ (p * a) := by rw [mul_comm, ← pow_mul],
+    simp only [expeq] }
+end
+
+/-- If `m : ℕ` is coprime with `n`,
+then the minimal polynomials of a primitive `n`-th root of unity `μ`
+has `μ ^ m` as root. -/
+lemma pow_is_root_minimal_polynomial {m : ℕ} (hcop : nat.coprime m n) :
+  is_root (map (int.cast_ring_hom K) (minimal_polynomial (is_integral h hpos))) (μ ^ m) :=
+begin
+  rw [minimal_polynomial_eq_pow_coprime h hpos hcop, is_root.def, eval_map],
+  have h := minimal_polynomial.aeval (is_integral (h.pow_of_coprime m hcop) hpos),
+  rwa [aeval_def (μ ^ m) _] at h
+end
+
+/-- `primitive_roots n K` is a subset of the roots of the minimal polynomial of a primitive
+`n`-th root of unity `μ`. -/
+lemma is_roots_of_minimal_polynomial : (primitive_roots n K).val ≤ (map (int.cast_ring_hom K)
+  (minimal_polynomial (is_integral h hpos))).roots :=
+begin
+  apply function.comp (multiset.le_iff_subset (primitive_roots n K).nodup).2 multiset.subset_iff.2,
+  intros x hx,
+  replace hx := (is_primitive_root_iff h hpos).1 ((mem_primitive_roots hpos).1 (finset.mem_def.2 hx)),
+  obtain ⟨m, hle, hcop, hx⟩ := hx,
+  rw [← hx, mem_roots _],
+  { exact pow_is_root_minimal_polynomial h hpos hcop },
+  { exact map_monic_ne_zero (minimal_polynomial.monic (is_integral h hpos)) }
+end
+
```

Comment:
If you use the statement `is_roots_of_minimal_polynomial'` I wrote above, and add somewhere near `polynomial.degree_map_le` the following lemma which seems to be missing:
```lean

lemma polynomial.nat_degree_map_le {R S : Type*} [semiring R] {p : polynomial R} [semiring S] (f : R →+* S) :
    (map f p).nat_degree ≤ p.nat_degree :=
begin
  by_cases hp : p = 0,
  { simp [hp] },
  { rw [← with_bot.coe_le_coe, ← degree_eq_nat_degree hp],
    by_cases hfp : map f p = 0,
    { simp [hfp, zero_le_degree_iff.mpr hp] },
    { simp [← degree_eq_nat_degree hfp, degree_map_le] } }
end
```
then you can get the following version of your last lemma:
```lean
/- The degree of the minimal polynomial of `μ` is at least `totient n`. -/
lemma totient_le_degree_minimal_polynomial : nat.totient n ≤ (minimal_polynomial
  (is_integral h hpos)).nat_degree :=
let P : polynomial ℤ := minimal_polynomial (is_integral h hpos),-- minimal polynomial of `μ`
    P_K : polynomial K := map (int.cast_ring_hom K) P -- minimal polynomial of `μ` sent to `K[X]`
in calc 
n.totient = (primitive_roots n K).card : (h.card_primitive_roots hpos).symm
... ≤ P_K.roots.to_finset.card : finset.card_le_of_subset (is_roots_of_minimal_polynomial' h hpos)
... ≤ P_K.roots.card : multiset.to_finset_card_le _ 
... ≤ P_K.nat_degree : (card_roots' $ map_monic_ne_zero (minimal_polynomial.monic $ is_integral h hpos))
... ≤ P.nat_degree : polynomial.nat_degree_map_le _
```